include/ddc/kernels/splines/knots_as_interpolation_points.hpp

```diff
@@ -31,26 +31,35 @@ class KnotsAsInterpolationPoints
 
     using tag_type = typename BSplines::tag_type;
 
+    template <class Sampling>
+    struct IntermediateUniformSampling
+        : UniformPointSampling<typename Sampling::continuous_dimension_type>
+    {
+    };
+
+    template <class Sampling>
+    struct IntermediateNonUniformSampling
+        : NonUniformPointSampling<typename Sampling::continuous_dimension_type>
+    {
+    };
+
```

Comment:
Is this needed in every file with a `get_sampling` method? Why?

Reply:
It is needed when you want to use an intermediate dimension (explaining this awesome name) that is used elsewhere which the case of `get_sampling`.

Reply:
I don't follow. If you need an intermediate dimension then don't you need to create a class such as this one? For Greville an intermediate class is needed so it is created and used but here the type is unused

Reply:
Ah yes I did not spot that. I will remove it